### ISSUE TITLE
Package causes Android release build of app to crash on startup

### ISSUE THREAD

**Issue by JediIzzyT** [gbwisx/msal_mobile#11]
Hello!

Please see [this](https://stackoverflow.com/questions/64126740/flutter-app-crash-on-startup-with-release-apk-but-runs-fine-with-flutter-run/65927932#65927932) for more information.

When using the packaged, if `minifyEnabled`, `shrinkResources`, or `useProguard` are not set to false in `android/app/build.gradle`, release builds of my app will crash on startup. I got the app to work by turning those off, but that isn't ideal. Is there a known fix for this?

Thanks!

**Comment by clragon** [comment]
in case you didnt fix this yet or you just arrived in this issue, I have found a solution for myself.

add a proguard rule file with the following rules:
```
-keep class com.microsoft.** { *; }
-keep class com.gbwisx.msal_mobile.** { *; }
```

hope it helps.

**Comment by JediIzzyT** [comment]
Thank you so much @clragon !! Your fix worked perfectly!

For those unfamiliar with the specific process described by @clragon , using Android Studio, in the android/app/build.gradle file, add this line:
```
android {
    ...
    buildTypes {
        ...
        proguardFiles getDefaultProguardFile('proguard-android.txt') , 'proguard-rules.pro' // Add this line
    }
}
```

Then, in the same directory, create a file named `proguard-rules.pro` and put in the following code:
```
-keep class com.microsoft.** { *; }
-keep class com.gbwisx.msal_mobile.** { *; }
```

Once those files are updated/created, right-click on the "android" folder in your Flutter app, navigate to the "Flutter" option, and select "Open Android module in Android Studio". I recommend opening it in a new window.

Once that window loads, click the "Sync Project with Gradle Files" button on the top right of the screen and wait for that process to run. You should be able to keep track of the progress on the very bottom of the window.

Finally, navigate to the "Build" option on the top toolbar, and click "Rebuild Project." Once that process has finished, you can close this window and publish your working app!

**Comment by clragon** [comment]
this would probably also fix #10

**Comment by clragon** [comment]
maybe the guide that @JediIzzyT wrote could be added to the readme, it would complement it nicely @gbwisx
